parquet/src/main/java/com/netflix/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -282,7 +285,7 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       }
 
       Set<T> dictionary = dict(id, lit.comparator());
-      if (dictionary.size() > 1) {
+      if (dictionary.size() > 1 || mayContainNulls.get(id)) {
```

Comment:
Nitpick - seeing that `EvalVisitor#dict` may return `null` you might consider returning a `java.util.Collections.emptySet()` instead [here](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/com/netflix/iceberg/parquet/ParquetDictionaryRowGroupFilter.java#L313)? It could save the trouble of having to NPE check on `dictionary` in all the methods where we have its size evaluated

Reply:
This is only run if the check above doesn't return. That check guarantees that all pages are dictionary-encoded so there must be a dictionary. I think it may be better to throw an exception in `dict` if there is no dictionary instead of handling dictionary-encoded pages with no dictionary here..

Reply:
I'm convinced 👍

Reply:
Want to open a PR to fix it?

Reply:
Will this one do? Should I open the PR against your branch here w/ the fixes? You might want to squash before merging yours though, right?

https://github.com/apache/incubator-iceberg/pull/93